src/ostree/ot-builtin-commit.c

```diff
@@ -636,11 +636,22 @@ ostree_builtin_commit (int argc, char **argv, OstreeCommandInvocation *invocatio
             {
               if (!opt_tar_pathname_filter)
                 {
-                  object_to_commit = g_file_new_for_path (tree);
-                  if (!ostree_repo_write_archive_to_mtree (repo, object_to_commit, mtree, modifier,
-                                                           opt_tar_autocreate_parents,
-                                                           cancellable, error))
-                    goto out;
+                  if (strcmp(tree, "-") == 0)
```

Comment:
I can make this change, but are you sure ? It seems to go against the style used throughout `ot-builtin-commit.c` which uses `strcmp()` throughout...

Reply:
Yeah, I like to promote `g_str_equal` when possible since it's exactly the same thing, but much more legible (the exception being if either can be `NULL`, in which case it's `g_strcmp0`, though that's not the case here). But meh, not worth another round trip!